src/time.c

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcutils/time.h"
+
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "rcutils/allocator.h"
+#include "rcutils/error_handling.h"
+#include "rcutils/snprintf.h"
+
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+rcutils_ret_t
+rcutils_time_point_value_as_nanoseconds_string(
+  const rcutils_time_point_value_t * timepoint,
+  char * str,
+  size_t str_size)
+{
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(timepoint, RCUTILS_RET_INVALID_ARGUMENT, allocator)
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(str, RCUTILS_RET_INVALID_ARGUMENT, allocator)
+  if (*timepoint >= 0) {
+    // have an explicit leading space to align with negative values which start with `-`
+    if (rcutils_snprintf(str, str_size, " %.19" PRId64, *timepoint) <= 0) {
```

Comment:
If `str_size` was specified as 1, the `snprintf`'s return value will be 0 from my understanding ("The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character."). I don't think we should error in that case anymore. However, there's [a test below](https://github.com/ros2/rcutils/pull/85/files#diff-885f7a53192e3dffac3511e600e2afbdR159) for `str_size=1` that expects `RCUTILS_RET_OK`, is that test passing?

Reply:
All of the tests are passing (and running 😄) from what I can tell. As well as on Windows where the `snprintf` like functions some times behave slightly differently.

I think that it will return `21` or whatever it _would_ have been, regardless of the value of `str_size`, rather than returning how many it _actually_ wrote based on truncating due to `str_size` being smaller than the result.

Reply:
ohhh, I see. it will successfully put (0 chars + NUL) but won't _return_ 0. good good, this clears it up, thanks